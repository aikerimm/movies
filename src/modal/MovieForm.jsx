import './movieModal.css';
import PropTypes from 'prop-types';

const MovieForm = ({
  movieTitle,
  releaseDate,
  movieUrl,
  rating,
  genre,
  runtime,
  overview,
}) => {
  return (
    <>
      <div className='formInput'>
        <label htmlFor='title'>Title</label>
        <input
          type='text'
          id='title'
          name='title'
          defaultValue={movieTitle}
        ></input>
      </div>
      <div className='formInput'>
        <label htmlFor='releaseDate'>Release Date</label>
        <input
          type='date'
          id='releaseDate'
          name='releaseDate'
          defaultValue={releaseDate ? releaseDate.toJSON().slice(0, 10) : null}
        ></input>
      </div>
      <div className='formInput'>
        <label htmlFor='movieUrl'>Movie Url</label>
        <input
          type='url'
          name='movieUrl'
          id='movieUrl'
          placeholder='https://'
          defaultValue={movieUrl}
        ></input>
      </div>
      <div className='formInput'>
        <label htmlFor='rating'>Rating</label>
        <input
          type='text'
          id='rating'
          name='rating'
          pattern='^[0-9]\.[0-9]$'
          title='Range: 0.0-9.9'
          defaultValue={rating}
        ></input>
      </div>
      <div className='formInput'>
        <label htmlFor='genre'>Genre</label>
        <select id='genre' name='genre' defaultValue={genre} multiple>
          <option value=''>Select Genre</option>
          <option value='Drama'>Drama</option>
          <option value='Comedy'>Comedy</option>
          <option value='Superhero'>Superhero</option>
          <option value='Crime'>Crime</option>
        </select>
      </div>
      <div className='formInput'>
        <label htmlFor='runtime'>Runtime</label>
        <input
          type='text'
          id='runtime'
          name='runtime'
          pattern='^[0-9]*'
          placeholder='minutes'
          title='Positive number'
          defaultValue={runtime}
        ></input>
      </div>
      <div className='formInput overviewInput'>
        <label htmlFor='overview'>Overview</label>
        <textarea
          id='overview'
          name='overview'
          rows='4'
          placeholder='Movie Description'
          defaultValue={overview}
        ></textarea>
      </div>
    </>
  );
};

MovieForm.propTypes = {
  movieTitle: PropTypes.string,
  releaseDate: PropTypes.instanceOf(Date),
  movieUrl: PropTypes.string,
  rating: PropTypes.number,
  genre: PropTypes.array,
  runtime: PropTypes.number,
  overview: PropTypes.string,
};

export default MovieForm;
