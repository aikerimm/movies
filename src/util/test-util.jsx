export const testMovie = {
  id: 123,
  title: 'Zootopia',
  vote_average: 9.8,
  runtime: 120,
  genres: ['Drama', 'Comedy'],
  release_date: '1994-01-01',
  overview: 'Lorem Ipsum Some',
  poster_path: '',
};