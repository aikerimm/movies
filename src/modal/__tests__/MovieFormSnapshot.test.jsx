/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer';
import MovieForm from '../MovieForm.jsx';
import { testMovie } from '../../util/test-util.jsx';

it('MovieForm renders correctly', () => {
  const onMovieFormSubmit = () => {
    console.log('test');
  };
  const tree = renderer
    .create(
      <MovieForm movie={testMovie} onMovieFormSubmit={onMovieFormSubmit} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
