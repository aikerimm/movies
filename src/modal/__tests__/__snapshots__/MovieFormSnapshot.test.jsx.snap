// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieForm renders correctly 1`] = `
<form
  className="addMovieForm"
  onSubmit={[Function]}
>
  <div
    className="formInput"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Movie Title"
      type="text"
      value="Zootopia"
    />
  </div>
  <div
    className="formInput"
  >
    <label
      htmlFor="release_date"
    >
      Release Date
    </label>
    <input
      id="release_date"
      name="release_date"
      onChange={[Function]}
      type="date"
      value="1994-01-01"
    />
  </div>
  <div
    className="formInput"
  >
    <label
      htmlFor="movieUrl"
    >
      Movie Url
    </label>
    <input
      id="poster_path"
      name="poster_path"
      onChange={[Function]}
      placeholder="https://"
      type="url"
      value=""
    />
  </div>
  <div
    className="formInput"
  >
    <label
      htmlFor="vote_average"
    >
      Rating
    </label>
    <input
      id="vote_average"
      name="vote_average"
      onChange={[Function]}
      placeholder="for ex. 9.9"
      type="text"
      value={9.8}
    />
  </div>
  <div
    className="formInput"
  >
    <label
      htmlFor="genres"
    >
      Genre
    </label>
    <select
      id="genres"
      multiple={true}
      name="genres"
      onChange={[Function]}
      value={
        [
          "Drama",
          "Comedy",
        ]
      }
    >
      <option
        value="Action"
      >
        Action
      </option>
      <option
        value="Adventure"
      >
        Adventure
      </option>
      <option
        value="Animation"
      >
        Animation
      </option>
      <option
        value="Comedy"
      >
        Comedy
      </option>
      <option
        value="Crime"
      >
        Crime
      </option>
      <option
        value="Drama"
      >
        Drama
      </option>
      <option
        value="Family"
      >
        Family
      </option>
      <option
        value="Fantasy"
      >
        Fantasy
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Horror"
      >
        Horror
      </option>
      <option
        value="Music"
      >
        Music
      </option>
      <option
        value="Mystery"
      >
        Mystery
      </option>
      <option
        value="Romance"
      >
        Romance
      </option>
      <option
        value="Science Fiction"
      >
        Science Fiction
      </option>
      <option
        value="Thriller"
      >
        Thriller
      </option>
      <option
        value="War"
      >
        War
      </option>
    </select>
  </div>
  <div
    className="formInput"
  >
    <label
      htmlFor="runtime"
    >
      Runtime
    </label>
    <input
      id="runtime"
      name="runtime"
      onChange={[Function]}
      placeholder="in minutes"
      type="text"
      value={120}
    />
  </div>
  <div
    className="formInput overviewInput"
  >
    <label
      htmlFor="overview"
    >
      Overview
    </label>
    <textarea
      id="overview"
      name="overview"
      onChange={[Function]}
      placeholder="Movie Description"
      rows="4"
      value="Lorem Ipsum Some"
    />
  </div>
  <div
    className="btnContainer"
  >
    <input
      className="cancelBtn roundedCorners"
      onClick={[Function]}
      type="button"
      value="Reset"
    />
    <button
      className="submitBtn roundedCorners"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
